library/core/src/primitive_docs.rs

```diff
@@ -1317,6 +1317,42 @@ mod prim_f16 {}
 /// | `wasm32`, `wasm64` | If all input NaNs are quiet with all-zero payload: None.<br> Otherwise: all possible payloads. |
 ///
 /// For targets not in this table, all payloads are possible.
+///
+/// # Algebraic operators
+///
+/// Algebraic operators of the form `a.algebraic_*(b)` allow the compiler to reassociate individual
+/// floating point operations for better vectorization. The exact set of optimizations is
```

Comment:
The first sentence says "reassociate" which points to a fairly specific transformation. Only later does it become clear that this is more general. I think we should word the first sentence so that if someone stops reading after that they don't have a completely wrong impression.
```suggestion
/// Algebraic operators of the form `a.algebraic_*(b)` allow the compiler to optimize
/// floating point operations using all the usual algebraic properties of real numbers --
/// despite the fact that those properties do *not* hold on floating point numbers.
/// This can give a great performance boost since it may unlock vectorization.
/// The exact set of optimizations is
```